src/avatar.h

```diff
@@ -2,56 +2,58 @@
 #ifndef CATA_SRC_AVATAR_H
 #define CATA_SRC_AVATAR_H
 
-#include <cstddef>
-#include <iosfwd>
+#include <array>
 #include <list>
 #include <map>
 #include <memory>
+#include <set>
 #include <string>
+#include <string_view>
 #include <unordered_set>
+#include <utility>
 #include <vector>
 
+#include "bodypart.h"
 #include "calendar.h"
 #include "character.h"
+#include "character_id.h"
 #include "coordinates.h"
 #include "enums.h"
 #include "game_constants.h"
+#include "item.h"
 #include "magic_teleporter_list.h"
 #include "mdarray.h"
-#include "memory_fast.h"
 #include "point.h"
 #include "type_id.h"
+#include "units.h"
 
 class advanced_inv_area;
 class advanced_inv_listitem;
 class advanced_inventory_pane;
 class cata_path;
 class diary;
 class faction;
-class item;
```

Comment:
I'm a bit concerned about the impact of losing some of these forward declarations. Avatar.h is included in a huge number of other places,  and pulling in a large header file like item.h does really bad things to compile times.

It's possible item.h and these other headers were already transitively included,  in which case nevermind.

Reply:
I had the same thought. I checked and IWYU is smart. They were already included transitively.

Reply:
I mean, it sometimes doesn't compile since it didn't require enough stuff, but I have never noticed it to include more stuff.